src/Microsoft.Identity.Client/TokenCache.Notifications.cs

```diff
@@ -137,5 +137,15 @@ public void SetBeforeWriteAsync(Func<TokenCacheNotificationArgs, Task> beforeWri
             AsyncBeforeWrite = beforeWrite;
         }
 #endif
+
+        private static void GuardOnMobilePlatforms()
+        {
+#if ANDROID || iOS
+        throw new PlatformNotSupportedException("You should not use these TokenCache methods object on mobile platforms. " +
+            "They meant to allow applications to define their own storage strategy on .net desktop and non-mobile platforms such as .net core. " +
+            "On mobile platforms, a secure and performant storage mechanism is implemeted by MSAL. " +
```

Comment:
typo: implemeted => implemented.

Maybe use the active voice: "On mobile platforms, MSAL.NET implements already a secure and performant storage mechanism."

Reply:
+1

---
In reply to: [286805957](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1170#discussion_r286805957) [](ancestors = 286805957)